src/backend/distributed/commands/alter_table.c

```diff
@@ -1199,7 +1199,12 @@ CreateDistributedTableLike(TableConversionState *con)
 	}
 
 	char partitionMethod = PartitionMethod(con->relationId);
-	CreateDistributedTable(con->newRelationId, newDistributionKey, partitionMethod,
+
+	char *distributionColumnName =
+		ColumnToColumnName(con->newRelationId,
+						   nodeToString(newDistributionKey));
```

Comment:
this nodeToString() thing is also quite weird, considering that the first thing that ColumnToColumnName does is stringToNode()

Reply:
considering to change it to `ColumnToColumnName(Oid, Node)`